### PR TITLE
Adding override and const to templates after recent changes in framework

### LINKED ISSUE #216: Updating class skeleton generators
Recently few changes happened that affect the REST process (or metadata) classes. It would be good that the macros

```
macros/REST_MakeMetadata.C
macros/REST_MakeProcess.C
```

would be updated accordingly, i.e. adding `override` and any other required update.

### PR DESCRIPTION
**Pull request by juanangp**
juanangp Ok: 14 [](https://gitlab.cern.ch/rest-for-physics/framework/-/commits/issue216)

Fixes #216

### CODE REVIEW

**Review by lobis — APPROVED** [review]

File: macros/REST_MakeProcess.C [hunk 1]
```diff
@@ -128,17 +128,17 @@ Int_t REST_MakeProcess(TString name, TString inputevent = "TRestEvent", TString
...
     headerfile << endl;
     headerfile << "public:" << endl;
-    headerfile << "    any GetInputEvent() { return " << inputeventname << "; }" << endl;
-    headerfile << "    any GetOutputEvent() { return " << outputeventname << "; }" << endl;
+    headerfile << "    any GetInputEvent() const override { return " << inputeventname << "; }" << endl;
+    headerfile << "    any GetOutputEvent() const override { return " << outputeventname << "; }" << endl;
     headerfile << endl;
-    headerfile << "    void InitProcess();" << endl;
+    headerfile << "    void InitProcess() override;" << endl;
     headerfile << endl;
```
Comment (jgalan) [inline macros/REST_MakeProcess.C:135]:
```suggestion
    headerfile << endl;
    headerfile << "const char* GetProcessName() override const { return \"CHANGEME!\"; }" << endl;
    headerfile << endl;
```

Comment (jgalan) [reply macros/REST_MakeProcess.C:135]: I think before `GetProcessName` was not pure virtual.

Comment (juanangp) [reply macros/REST_MakeProcess.C:135]: That's weird, I have compiled a dummy library in the code without this line and without any isse.

Btw, `override` should be placed after the `const` --> `const char* GetProcessName() const override`

Comment (jgalan) [reply macros/REST_MakeProcess.C:135]: Thats surprising, sure it was a process?

Comment (juanangp) [reply macros/REST_MakeProcess.C:135]: Apparently it gives runtime error but not compilation error...

**Review by jgalan — APPROVED** [review]

**Pull request merged**